src/github.com/getlantern/flashlight/app/app.go

```diff
@@ -49,6 +50,16 @@ func (app *App) Init() {
 	// use buffered channel to avoid blocking the caller of 'AddExitFunc'
 	// the number 10 is arbitrary
 	app.chExitFuncs = make(chan func(), 10)
+
+	// I'm actually unclear why we can't just send ui.RegisterType directly to
+	// newNotifications here, but the compiler doesn't like it for some reason.
+	register := func(t string) (notify.UISender, error) {
```

Comment:
I think that's because they're different function types. Here you're casting the function's output into `notify.UISender, error`, and not trying to cast a whole function type into another.